packages/cf-component-input/src/Input.js

```diff
@@ -1,22 +1,38 @@
 import React from 'react';
-
 import PropTypes from 'prop-types';
+import { createComponent } from 'cf-style-container';
 
-class Input extends React.Component {
-  render() {
-    let { className, type, invalid, ...props } = this.props;
+const styles = ({ theme }) => ({
+  zIndex: theme.zIndex,
+  width: theme.width,
+  margin: theme.margin,
+  padding: theme.padding,
+  border: theme.border,
+  borderRadius: theme.borderRadius,
+
+  verticalAlign: theme.verticalAlign,
+  fontFamily: theme.fontFamily,
+  fontSize: theme.fontSize,
 
-    let _className = 'cf-input cf-input--' + type;
+  background: theme.background,
+  color: theme.color,
+  outline: theme.outline,
 
-    if (invalid) {
-      _className += ' cf-input--invalid';
-    }
+  transition: theme.transition,
 
-    if (className) {
-      _className += ' ' + className;
-    }
+  '&:hover': {
+    borderColor: theme['&:hover'].borderColor
+  },
 
-    return <input className={_className} type={type} {...props} />;
+  '&:focus': {
+    outline: theme['&:hover'].outline,
+    outlineOffset: theme['&:hover'].outlineOffset
+  }
+});
+
+class Input extends React.Component {
+  render() {
+    return <input {...this.props} />;
```

Comment:
I'm not sure. I think I'd still want to, to be able to add things like `aria-*` classes and such? I remember needing it at a certain point but maybe that was just that `className` debate we had and doesn't apply anymore since we have fela?